src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -718,6 +783,21 @@ let init_from_config_file ?(genesis_dir = Cache_dir.autogen_path) ~logger
           "Proof level %s is not compatible with compile-time proof level %s"
           (str proof_level) (str compiled)
   in
+  let%bind genesis_ledger, ledger_config, ledger_file =
+    Ledger.load ~genesis_dir ~logger ~constraint_constants
+      (Option.value config.ledger
+         ~default:
+           { base= Named Coda_compile_config.genesis_ledger
+           ; num_accounts= None
+           ; hash= None })
+  in
+  let config =
+    {config with ledger= Option.map config.ledger ~f:(fun _ -> ledger_config)}
+  in
+  let%bind genesis_constants =
```

Comment:
could you use vanilla `let` here by omitting the `Deferred.return`?

Reply:
The underlying monad is `Deferred.Or_error.t`, so this is to lift the `Or_error.t` to that. Leaving this, couldn't come up with anything tidier.